File: backend/src/main/java/com/affairs/management/entity/Department.java
package com.affairs.management.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 部门实体
 */
@Data
public class Department {
    /** 部门ID，如D001 */
    private String id;
    /** 部门名称 */
    private String name;
    /** 上级部门ID */
    private String parentId;
    /** 部门负责人用户ID */
    private String leaderId;
    /** 排序 */
    private Integer sortOrder;
    /** 层级 */
    private Integer level;
    /** 成员数量（冗余） */
    private Integer memberCount;
    /** 状态：0=停用 1=启用 */
    private Integer status;
    /** 创建时间 */
    private LocalDateTime createdAt;
    /** 更新时间 */
    private LocalDateTime updatedAt;
}

File: src/types/department.ts
// ============================================
// 部门 & 组织架构 相关类型
// ============================================

/** 部门 */
export interface Department {
  id: string
  name: string
  parentId: string | null
  leaderId: string
  leaderName: string
  sort: number
  level: number
  memberCount: number
  status: 0 | 1
  createdAt: string
  updatedAt: string
  children?: Department[]
}

/** 组织架构树节点 */
export interface OrgTreeNode {
  id: string
  label: string
  type: 'dept' | 'member'
  parentId: string | null
  leaderName?: string
  memberCount?: number
  position?: string
  avatar?: string
  children?: OrgTreeNode[]
}

/** 创建/编辑部门请求 */
export interface DeptFormData {
  name: string
  parentId: string | null
  leaderId: string
  sort: number
  status?: 0 | 1
}

/** 人员列表查询参数 */
export interface MemberListParams {
  deptId?: string
  keyword?: string
  position?: string
  page: number
  pageSize: number
}
